video/mkv_decoder.cpp

```diff
@@ -238,14 +237,18 @@ bool MKVDecoder::loadStream(Common::SeekableReadStream *stream) {
 	_pBlock = _pBlockEntry->GetBlock();
 	_trackNum = _pBlock->GetTrackNumber();
 	_frameCount = _pBlock->GetFrameCount();
+	_fileStream = stream;
 
 	return true;
 }
 
 void MKVDecoder::close() {
 	VideoDecoder::close();
 
+	delete _fileStream;
```

Comment:
To be really clean, reader should be deleted before as it's created after and depends on fileStream